internal/client.go

```diff
@@ -1003,6 +1027,32 @@ func DialCloudOperationsClient(ctx context.Context, options CloudOperationsClien
 	}, nil
 }
 
+// TODO
+func PrepareUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
+	input, err := createUpdateWorkflowInput(options)
+	if err != nil {
+		return nil, err
+	}
+	return &UpdateWorkflowOperation{input: input}, nil
+}
+
+// Get returns a WorkflowUpdateHandle, or an error if the update request failed.
+// TODO
+func (op *UpdateWorkflowOperation) Get(run WorkflowRun) (WorkflowUpdateHandle, error) {
+	if run == nil {
+		return nil, errors.New("invalid workflow run: is nil")
+	}
+	if runImpl, ok := run.(*workflowRunImpl); ok {
+		if runImpl.startRequest != op.startRequest {
+			return nil, errors.New("invalid workflow run: does not belong to update operation")
+		}
+	}
+	return op.interceptor.updateHandleFromResponse(
+		context.Background(), enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_UNSPECIFIED, op.resp)
```

Comment:
By using wait stage unspecified, it returns a handle (either completed or lazy) immediately without blocking.